From f86305fc0d4c4b918d81f36c9530bafe0e769e94 Mon Sep 17 00:00:00 2001
From: Ruslan Bay <ruslanbey@proton.me>
Date: Mon, 9 Mar 2026 23:18:59 +0100
Subject: [PATCH 19/27] arm64: dts: exynos: add starlte thermal zones

---
 .../boot/dts/exynos/exynos9810-starlte.dts    | 21 +++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts b/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts
index fb63cd32336d..7b0435faa289 100644
--- a/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts
+++ b/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts
@@ -74,6 +74,27 @@ memory@80000000 {
 		      <0x8 0x80000000 0x80000000>;
 	};
 
+	thermal-zones {
+		cpu-thermal {
+			polling-delay-passive = <250>;
+			polling-delay = <1000>;
+
+			trips {
+				cpu_alert0: cpu-alert0 {
+					temperature = <75000>;
+					hysteresis = <2000>;
+					type = "passive";
+				};
+
+				cpu_crit0: cpu-crit0 {
+					temperature = <115000>;
+					hysteresis = <0>;
+					type = "critical";
+				};
+			};
+		};
+	};
+
 	reserved-memory {
 		#address-cells = <2>;
 		#size-cells = <1>;
-- 
2.53.0

