From b74f4c9fba48b2b948e95550f683316ab2619b04 Mon Sep 17 00:00:00 2001
From: Ruslan Bay <ruslanbey@proton.me>
Date: Mon, 9 Mar 2026 23:19:00 +0100
Subject: [PATCH 20/28] arm64: dts: exynos: starlte: add Wi-Fi/BT board wiring

---
 .../boot/dts/exynos/exynos9810-starlte.dts    | 24 +++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts b/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts
index 37fdfc70b..9b1438131 100644
--- a/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts
+++ b/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts
@@ -248,6 +248,18 @@ &ufs_phy {
 	status = "okay";
 };
 
+/ {
+	bluetooth {
+		compatible = "brcm,bcm4329-bt";
+		shutdown-gpios = <&gpg2 0 GPIO_ACTIVE_HIGH>;
+		device-wakeup-gpios = <&gpg2 0 GPIO_ACTIVE_HIGH>;
+		host-wakeup-gpios = <&gpa2 3 GPIO_ACTIVE_HIGH>;
+		pinctrl-0 = <&bt_hostwake &bt_btwake>;
+		pinctrl-names = "default";
+		status = "disabled";
+	};
+};
+
 &mmc {
 	pinctrl-0 = <&sd2_clk &sd2_cmd &sd2_bus1 &sd2_bus4 &sd2_cd>;
 	pinctrl-names = "default";
@@ -268,6 +280,18 @@ &mmc {
 	samsung,dw-mshc-sdr-timing = <0 3>;
 
 	status = "okay";
+
+	wifi@1 {
+		compatible = "brcm,bcm4359-fmac", "brcm,bcm4329-fmac";
+		reg = <1>;
+		interrupt-parent = <&gpa0>;
+		interrupts = <7 IRQ_TYPE_LEVEL_LOW>;
+		interrupt-names = "host-wake";
+		reset-gpios = <&gpg0 5 GPIO_ACTIVE_LOW>;
+		pinctrl-0 = <&cfg_wlanen &wlan_host_wake>;
+		pinctrl-names = "default";
+		status = "disabled";
+	};
 };
 
 &usbdrd {
-- 
2.53.0

